src/shogun/machine/KernelMachine.cpp

```diff
@@ -452,8 +452,10 @@ bool CKernelMachine::train_locked(SGVector<index_t> indices)
 
 	/* dont do train because model should not be stored (no acutal features)
 	 * and train does data_unlock */
+	auto sub = connect_to_signal_handler();
```

Comment:
I wonder whether we shouldnt do this in the base class? I.e. `CMachine::train` ?
This way we need to register every machine specialization. Or do I see this wrong?

Reply:
It is already implemented into CMachine:train. However, there are some classes which implement their own version of the train method and therefore they do not have the connection with the signal handler.

Reply:
ok, as discussed, let's try to move this into the base class and for the spezializations that overload the train method can just call the base class (e.g. in CLinearMachine)